test/service/test_response.py

```diff
@@ -2654,76 +2680,112 @@ def tearDownClass(cls):
+                    response = requests.get(str(url))
+                    response.raise_for_status()
+                    response = response.json()
+                    actual = list(map(CellCounts.from_response, response['hits']))
+                    if not ascending:
+                        expected = list(reversed(expected))
+                    self.assertEqual(expected, actual)
 
     def test_filter_by_cell_count(self):
         """
-        Verify filtering by 'cellCount' filters the documents based on the total
-        number of cells in each document, using the sum of total cells when a
-        document contains more than one cell suspension inner entity.
+        Verify filtering projects by the total cell suspension cell count, the
+        project cell count, and the effective cell count.
         """
-        params = {
-            'catalog': self.catalog,
-            'filters': json.dumps({
-                'cellCount': {
-                    'within': [
-                        [
-                            6000,
-                            9000
-                        ]
-                    ]
-                }
-            })
+        test_cases = {
+            'cellCount': {
+                None: [],
+                6210: [
+                    CellCounts(None, {'Brain': 6210}),
+                ],
+                (3000, 8000): [
+                    CellCounts(None, {'Brain': 6210}),
+                    CellCounts(None, {'presumptive gut': 3869, 'endoderm': 3869}),
+                ],
+            },
+            'projectEstimatedCellCount': {
+                None: [
+                    CellCounts(None, {'Brain': 6210}),
+                    CellCounts(None, {'presumptive gut': 3869, 'endoderm': 3869}),
+                    CellCounts(None, {'brain': 10_000})
+                ],
+                3589: [
+                    CellCounts(3589, {'brain': 1}),
+                ],
+                (6000, 100_000): [
+                    CellCounts(88_000, {'mouth mucosa': None}),
+                ],
+            },
+            'effectiveCellCount': {
+                None: [],
+                10_000: [
+                    CellCounts(None, {'brain': 10_000})
+                ],
+                (3000, 7000): [
+                    CellCounts(3589, {'brain': 1}),
+                    CellCounts(None, {'Brain': 6210}),
+                ],
+            },
         }
-        url = self.base_url.set(path='/index/projects', args=params)
-        response = requests.get(str(url))
-        response.raise_for_status()
-        response_json = response.json()
-        actual_results = self._count_total_cells(response_json)
-        expected_results = [
-            (1, 6210),
-            (2, 7738)
-        ]
-        self.assertEqual(actual_results, expected_results)
+        for field, test_case in test_cases.items():
+            for filter_, expected in test_case.items():
```

Comment:
```suggestion
            for filter, expected in test_case.items():
```

If you get a warning about shadowing a built-in, disable that inspection globally.